src/zenml/models/pipeline_run_models.py

```diff
@@ -121,19 +136,48 @@ class PipelineRunResponseModel(
     stack: Optional["StackResponseModel"] = Field(
         default=None, title="The stack that was used for this run."
     )
-
     metadata: Dict[str, "RunMetadataResponseModel"] = Field(
         default={},
         title="Metadata associated with this pipeline run.",
     )
-
     build: Optional["PipelineBuildResponseModel"] = Field(
         default=None, title="The pipeline build that was used for this run."
     )
-
     deployment: Optional["PipelineDeploymentResponseModel"] = Field(
         default=None, title="The deployment that was used for this run."
     )
+    steps: Dict[str, "StepRunResponseModel"] = Field(
```

Comment:
Not wrong, let me explain 😄 

I discussed hydration a lot with @AlexejPenner and @Cahllagerfeld last week and thought about this quite a lot. My understanding so far is that:
- Having some hydration makes sense because it reduces the amount of API requests that the frontend needs to send. Especially waterfall-like request dependencies should generally be avoided as they can slow down the frontend a lot (which would be the case with, e.g., pipeline -> run -> step -> artifact -> metadata)
- Hydrating everything (as we currently almost do) is also bad for performance because you send a lot of redundant/useless data with each response

Therefore, I think we need to find a sensible middle ground. My rule of thumb so far is this: **For each dashboard page, there should be one or more API endpoints that you can query *simultaneously* to get all the information needed to display the page without requiring follow-up requests. All other hydration should be removed.**
- E.g.: ✅  runs detail page contains the pipeline name, which is in the pipeline model, so hydrating the pipeline model into the run probably makes sense
- E.g.: ❌ the overview page can query runs, stacks, components, pipelines, active user, ... all in parallel, so hydrating all of them into a mega overview model is not necessary

Applying this rule, `run.steps` hydration makes sense to me since the run DAG needs information from the run, each step, and each artifact. `step.outputs` is already hydrated into the step, so `run.steps` is the minimal hydration that needs to be added to support fetching all information of the DAG with a single request (GET pipeline run).

Hope that clarifies it a bit 😄

Reply:
I guess I still don't understand the part where the dashboard can query the active user, as instead it would probably have to query the user that started the run? And that is only possible once the run response arrived to get the user ID.

But the entire hydration story is probably part of a bigger refactoring, so nothing to worrty about in this PR 👍

Reply:
Yeah the hydration story is for next week ;)